packages/monaco/src/browser/textmate/textmate-tokenizer.ts

```diff
@@ -57,44 +57,17 @@ export namespace TokenizerOption {
     };
 }
 
-export function createTextmateTokenizer(grammar: IGrammar, options: TokenizerOption): monaco.languages.TokensProvider {
+export function createTextmateTokenizer(grammar: IGrammar, options: TokenizerOption): monaco.languages.EncodedTokensProvider {
     if (options.lineLimit !== undefined && (options.lineLimit <= 0 || !Number.isInteger(options.lineLimit))) {
         throw new Error(`The 'lineLimit' must be a positive integer. It was ${options.lineLimit}.`);
     }
     return {
         getInitialState: () => new TokenizerState(INITIAL),
-        tokenize(line: string, state: TokenizerState) {
-            if (options.lineLimit !== undefined && line.length > options.lineLimit) {
-                // Line is too long to be tokenized
-                return { tokens: [], endState: state };
-            }
-            const result = grammar.tokenizeLine(line, state.ruleStack);
-            const tokenTheme = monaco.services.StaticServices.standaloneThemeService.get().getTheme().tokenTheme;
-            const defaultResult = tokenTheme.match(undefined, 'should.return.default');
-            const defaultForeground = monaco.modes.TokenMetadata.getForeground(defaultResult);
+        tokenizeEncoded(line: string, state: TokenizerState) {
+            const result = grammar.tokenizeLine2(line, state.ruleStack);
```

Comment:
I'm afraid the `lineLimit` checks from the previous `tokenize()` function should still be performed, otherwise we can end up with a frozen UI while tokenizing very long lines.